pom.xml

```diff
@@ -175,7 +182,7 @@
               <goal>copy-resources</goal>
             </goals>
             <configuration>
-              <outputDirectory>${project.build.directory}/${project.artifactId}/js</outputDirectory>
+              <outputDirectory>${project.basedir}/src/main/webapp/js</outputDirectory>
```

Comment:
`maven-hpi-plugin` cannot find resources if they are in the build directory. They need to be in the webapp directory to be picked up.

It works on a full winstone server but not the jetty dev server.

This also breaks SNAPSHOT dependencies across different plugins, e.g. creating a SNAPSHOT of this plugin and running it in the design library did not work for me

I think there's similar issues throughout the echarts / bootstrap plugins

Reply:
Unsure the impacts of changing it or if it supports multiple paths etc.

This is where the code is I believe:
https://github.com/jenkinsci/maven-hpi-plugin/blob/a041c9f3ec9675ae0ae3b56b3aa49a3c3ab7f089/src/main/java/org/jenkinsci/maven/plugins/hpi/HplMojo.java#L97-L98

> degenerates the developer experience in IntelliJ.
Without this fix the developer experience doesn't work at all in IntelliJ it's 100% broken.

I assume you use that docker dev thing here without `hpi:run`? (that's not how plugins are normally developed though)

I assume your complaint is around generated resources showing up in search, if we can't fix `maven-hpi-plugin` safely then we could likely include some config in this repo to mark it as excluded.